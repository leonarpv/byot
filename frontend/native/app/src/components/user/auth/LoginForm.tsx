import * as React from 'react';
import {ScrollView, StyleSheet} from 'react-native';
import {TextField} from '../../elements/lib/TextField';
import {Formik} from 'formik';
import {Text, View} from 'react-native-ui-lib';
import {useTranslation} from 'react-i18next';
import {loginSchema} from '@byot-frontend/common/src/types/schemas/validation/LoginSchema';
import {Button} from '../../elements/lib/Button';

export const testIDs = {
  root: 'loginForm-root',
  usernameEmail: 'loginForm-usernameEmail-input',
  password: 'loginForm-password-input',
  submitButton: 'loginForm-submitButton',
};

interface Props {
  onSubmit: (credentials: {usernameOrEmail: string; password: string}) => void;
  children?: React.ReactNode;
  loading?: boolean;
}

interface State {}

const makeStyles = (props: Props, state: State) =>
  StyleSheet.create({
    root: {
      flexDirection: 'column',
      alignItems: 'center',
    },
    form: {
      maxWidth: 400,
      width: '100%',
    },
    password: {
      marginBottom: 48,
    },
  });

export const LoginForm: React.FC<Props> = (props: Props) => {
  const styles = makeStyles(props, {});
  const {t} = useTranslation();
  return (
    <View accessibilityLabel={testIDs.root} testID={testIDs.root}>
      <Formik
        validationSchema={loginSchema(t)}
        validateOnBlur
        validateOnChange
        initialValues={{usernameOrEmail: '', password: ''}}
        onSubmit={c => props.onSubmit(c)}>
        {({handleChange, handleBlur, handleSubmit, values, touched, errors}) => (
          <>
            <Text heading>{t('Login')!}</Text>
            <ScrollView horizontal={false} keyboardShouldPersistTaps="handled">
              <View marginB-4>
                <TextField
                  placeholder={t('Username or email')}
                  value={values.usernameOrEmail}
                  onBlur={handleBlur('usernameOrEmail')}
                  onChangeText={handleChange('usernameOrEmail')}
                  keyboardType="email-address"
                  infoText
                  error={(touched.usernameOrEmail && errors.usernameOrEmail) as string}
                  testID={testIDs.usernameEmail}
                  accessibilityLabel={testIDs.usernameEmail}
                  autoCapitalize="none"
                />
              </View>
              <View marginB-4>
                <TextField
                  placeholder={t('Enter password')}
                  value={values.password}
                  secureTextEntry
                  onBlur={handleBlur('password')}
                  onChangeText={handleChange('password')}
                  style={styles.password}
                  infoText
                  error={(touched.password && errors.password) as string}
                  testID={testIDs.password}
                  accessibilityLabel={testIDs.password}
                  autoCapitalize="none"
                />
              </View>
              <Button
                testID={testIDs.submitButton}
                loading={props.loading}
                onPress={handleSubmit}
                label={t('Login')}
              />
              {props.children}
            </ScrollView>
          </>
        )}
      </Formik>
    </View>
  );
};
