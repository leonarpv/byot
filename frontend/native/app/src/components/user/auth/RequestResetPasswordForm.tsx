import * as React from 'react';
import {Formik} from 'formik';
import {useTranslation} from 'react-i18next';
import {requestResetPasswordSchema} from '@byot-frontend/common/src/types/schemas/validation/RequestResetPasswordSchema';
import {Text, View} from 'react-native-ui-lib';
import {TextField} from '@byot-frontend/native-app/src/components/elements/lib/TextField';
import {Button} from '../../elements/lib/Button';

export const testIDs = {
  email: 'requestResetPasswordForm-email-input',
};

interface Props {
  loading?: boolean;
  onSubmit: (email: string) => void;
}

export const RequestResetPasswordForm: React.FC<Props> = (props: Props) => {
  const initialValues = {email: ''};
  const {t} = useTranslation();
  return (
    <>
      <Text heading>{t('Login')!}</Text>
      <Formik
        initialValues={initialValues}
        validateOnBlur
        validateOnChange
        validationSchema={requestResetPasswordSchema(t)}
        onSubmit={({email}) => props.onSubmit(email)}>
        {({values, handleSubmit, handleChange, handleBlur, errors, touched}) => (
          <>
            <View marginB-4>
              <TextField
                placeholder={t('Enter email')}
                value={values.email}
                onBlur={handleBlur('email')}
                onChangeText={handleChange('email')}
                keyboardType="email-address"
                infoText
                error={(touched.email && errors.email) as string}
                testID={testIDs.email}
                accessibilityLabel={testIDs.email}
                autoCapitalize="none"
              />
            </View>
            <Button loading={props.loading} onPress={handleSubmit} label={t('Request reset password')} />
          </>
        )}
      </Formik>
    </>
  );
};
