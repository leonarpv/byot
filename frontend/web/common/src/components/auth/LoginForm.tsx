import * as React from 'react';
import {Input} from '../elementary/form/Input';
import {useTranslation} from 'react-i18next';
import {Button} from '../elementary/form/Button';
import {Form, Formik} from 'formik';
import * as Yup from 'yup';
import {TFunction} from 'i18next';
import {UserLogin} from '@byot-frontend/common/src/shared/graphql/ts/types';
import {PlainLayoutNarrow} from '../plain-layout/PlainLayoutNarrow';
import {PlainLayoutTitle} from '../plain-layout/PlainLayoutTitle';

type Values = UserLogin;
const loginSchema = (t: TFunction) =>
  Yup.object().shape({
    userNameOrEmail: Yup.string().required(t('Enter username or email')),
    password: Yup.string().required(t('Enter password')),
  });

interface Props {
  onLogin: (credentials: Values) => void;
  children?: React.ReactNode;
  loading?: boolean;
}

export const LoginForm: React.FC<Props> = (props: Props) => {
  const {t} = useTranslation();

  return (
    <PlainLayoutNarrow>
      <PlainLayoutTitle>{t('Login')}</PlainLayoutTitle>
      <Formik
        initialValues={{userNameOrEmail: '', password: ''}}
        onSubmit={values => props.onLogin(values)}
        validateOnChange
        validateOnBlur
        validationSchema={loginSchema(t)}>
        {({values, errors, touched, handleChange, handleBlur}) => (
          <Form data-testid="common-auth-login-form-form">
            <Input
              color="secondary"
              type="text"
              label={t('Username or email')}
              value={values.userNameOrEmail}
              onBlur={handleBlur('userNameOrEmail')}
              onChange={handleChange('userNameOrEmail')}
              name="userNameOrEmail"
              errorText={touched.userNameOrEmail && errors.userNameOrEmail}
              data-testid="common-auth-login-form-username"
            />
            <Input
              color="secondary"
              type="password"
              label={t('Password')}
              value={values.password}
              onBlur={handleBlur('password')}
              onChange={handleChange('password')}
              name="password"
              errorText={touched.password && errors.password}
              data-testid="common-auth-login-form-password"
            />
            <Button
              type="submit"
              color="gradient"
              data-testid="common-auth-login-form-button"
              fullWidth
              loading={props.loading}>
              {t('Login')}
            </Button>
            {props.children}
          </Form>
        )}
      </Formik>
    </PlainLayoutNarrow>
  );
};
