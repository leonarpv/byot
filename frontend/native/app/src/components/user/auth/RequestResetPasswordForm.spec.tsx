import * as React from 'react';
import {fireEvent, render, wait} from '@testing-library/react-native';
import {RequestResetPasswordForm, testIDs} from './RequestResetPasswordForm';
import {LoginForm} from './LoginForm';

describe('<RequestResetPasswordForm/>', () => {
  it('should submit form with valid email', () => {
    const email = 'email@email.sk';
    const onSubmit = jest.fn();
    const {asJSON, getByTestId} = render(<RequestResetPasswordForm onSubmit={onSubmit} />);
    const emailInput = getByTestId(testIDs.email)!;
    const button = getByTestId(testIDs.submit)!;
    fireEvent.changeText(emailInput, email);
    fireEvent.press(button);
    wait(() => expect(onSubmit).toBeCalledWith(email));
    expect(asJSON()).toMatchSnapshot();
  });
  it('should not submit form with invalid email', () => {
    const email = 'email@email.sk';
    const onSubmit = jest.fn();
    const {asJSON, getByTestId} = render(<RequestResetPasswordForm onSubmit={onSubmit} />);
    const emailInput = getByTestId(testIDs.email)!;
    const button = getByTestId(testIDs.submit)!;
    fireEvent.changeText(emailInput, email);
    fireEvent.press(button);
    wait(() => expect(onSubmit).not.toBeCalled());
    expect(asJSON()).toMatchSnapshot();
  });
});
