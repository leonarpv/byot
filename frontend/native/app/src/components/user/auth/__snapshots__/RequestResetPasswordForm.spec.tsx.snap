// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RequestResetPasswordForm/> should not submit form with invalid email 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <Text
    heading={true}
  >
    Request reset password
  </Text>
  <View
    marginB-4={true}
  >
    <View
      paddingB-8={true}
      paddingL-12={true}
      paddingR-12={true}
      paddingT-8={true}
      style={
        Array [
          Object {
            "backgroundColor": "#FAFAF1",
            "borderRadius": 8,
          },
          undefined,
        ]
      }
    >
      <TouchableWithoutFeedback>
        <TextInput
          accessibilityLabel="requestResetPasswordForm-email-input"
          allowFontScaling={true}
          autoCapitalize="none"
          color="#FAFAF1"
          enableErrors={false}
          floatOnFocus={true}
          floatingPlaceholder={true}
          floatingPlaceholderColor={
            Object {
              "default": "#FAFAF1",
              "error": "#FAFAF1",
              "focus": "#FAFAF1",
            }
          }
          floatingPlaceholderStyle={Object {}}
          hideUnderline={true}
          infoText={true}
          keyboardType="email-address"
          placeholder="Enter email"
          rejectResponderTermination={true}
          style={
            Array [
              undefined,
              Object {
                "marginBottom": 4,
                "marginTop": -4,
              },
            ]
          }
          testID="requestResetPasswordForm-email-input"
          underlineColorAndroid="transparent"
          value="email@email.sk"
        />
      </TouchableWithoutFeedback>
    </View>
    <Text
      error={false}
      style={
        Object {
          "fontSize": 12,
          "marginTop": 2,
          "paddingLeft": 12,
        }
      }
      success={false}
    >
       
    </Text>
  </View>
  <TouchableOpacity
    accessibilityLabel="element-button"
    label="Request reset password"
    testID="element-button"
  />
</View>
`;

exports[`<RequestResetPasswordForm/> should submit form with valid email 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <Text
    heading={true}
  >
    Request reset password
  </Text>
  <View
    marginB-4={true}
  >
    <View
      paddingB-8={true}
      paddingL-12={true}
      paddingR-12={true}
      paddingT-8={true}
      style={
        Array [
          Object {
            "backgroundColor": "#FAFAF1",
            "borderRadius": 8,
          },
          undefined,
        ]
      }
    >
      <TouchableWithoutFeedback>
        <TextInput
          accessibilityLabel="requestResetPasswordForm-email-input"
          allowFontScaling={true}
          autoCapitalize="none"
          color="#FAFAF1"
          enableErrors={false}
          floatOnFocus={true}
          floatingPlaceholder={true}
          floatingPlaceholderColor={
            Object {
              "default": "#FAFAF1",
              "error": "#FAFAF1",
              "focus": "#FAFAF1",
            }
          }
          floatingPlaceholderStyle={Object {}}
          hideUnderline={true}
          infoText={true}
          keyboardType="email-address"
          placeholder="Enter email"
          rejectResponderTermination={true}
          style={
            Array [
              undefined,
              Object {
                "marginBottom": 4,
                "marginTop": -4,
              },
            ]
          }
          testID="requestResetPasswordForm-email-input"
          underlineColorAndroid="transparent"
          value="email@email.sk"
        />
      </TouchableWithoutFeedback>
    </View>
    <Text
      error={false}
      style={
        Object {
          "fontSize": 12,
          "marginTop": 2,
          "paddingLeft": 12,
        }
      }
      success={false}
    >
       
    </Text>
  </View>
  <TouchableOpacity
    accessibilityLabel="element-button"
    label="Request reset password"
    testID="element-button"
  />
</View>
`;
